### PR TITLE
[MDS-6014] Spatial File FE

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
- show spatial files in a table
- upload spatial files in a modal
   - validate on FE
- show the geomark on the map in a view modal

## Misc. changes:
- change to map component to show geojson _features_
   - there were some issues with the timing of the call to `fitBounds` and with no minimum height set _in_ the component, issues rendering without  a container (see in action: look at the legal land owner page on the form on test, and delete the other elements in the same row. With no sibling elements, it will disappear)
- modal wrapper/add party component form: upgraded to tsx and props tidied up (there were unused props)
   - extracted the main modal logic to a common component, used to wrap around separate CORE/MS implementations 
- RenderFileUpload: dealt with an extension having the potential for multiple MIME-types. Cleared up variable ambiguity (lots of eslint warnings about reusing names). Destructured props, deleted unused, and attached the prop type to the component. Fixed bug where uppercase extensions weren't interpreted as valid. Disallowed empty files (everywhere that the common component is used). But a major takeaway here is that **RenderFileUpload will now yell at you if you give it bad props**
- I initially considered using the DocumentTable. That didn't go very well, but I did take out the bit where DocumentTable took DocumentTable as a prop.

[MDS-6014](https://bcmines.atlassian.net/browse/MDS-6014)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: services/common/src/redux/createAppSlice.ts [hunk 1]
```diff
@@ -4,3 +4,8 @@ import { buildCreateSlice, asyncThunkCreator } from "@reduxjs/toolkit";
 export const createAppSlice = buildCreateSlice({
   creators: { asyncThunk: asyncThunkCreator },
 });
+
+export const rejectHandler = (action) => {
+  console.log(action.error);
+  console.log(action.error.stack);
```
Comment (simensma-fresh) [inline services/common/src/redux/createAppSlice.ts:10]: Intentional to leave the console logs here?

File: services/document-manager/backend/Dockerfile [hunk 2]
```diff
@@ -17,4 +17,6 @@ EXPOSE 5001
 
 RUN mkdir -p /tmp/celery
 
+RUN mkdir -p /tmp/spatial
```
Comment (simensma-fresh) [inline services/document-manager/backend/Dockerfile]: I see that this doesn't match exactly the folder the assets are downloaded to. We should probably add the missing `/` to the docman paths

image

**Review by simensma-fresh — DISMISSED** [review]
Loving all the cleanup! 🧹 

Two small comments, but looks great!
(and a merge conflict)

File: services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    },
+  ];
+
+  const isFinalStep = currentStep === stepContent.length - 1;
+
+  return (
+    <FormWrapper
+      isModal
+      name={modalFormName}
+      onSubmit={async (values) => {
```
Comment (matbusby-fw) [inline services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx]: Not a huge deal, but I this function is getting a bit large to live inline.  Might be nice as it's own function.

File: services/common/src/components/forms/RenderFileUpload.tsx [hunk 4]
```diff
@@ -67,44 +64,54 @@ interface FileUploadProps extends BaseInputProps {
...
 
   beforeAddFile?: (file?: any) => any;
   beforeDropFile?: (file?: any) => any;
   onProcessFiles?: () => void;
-  onRemoveFile?: () => void;
+  onRemoveFile?: (error, file) => void;
   addFileStart?: () => void;
 }
 
-const defaultProps = {
```
Comment (matbusby-fw) [inline services/common/src/components/forms/RenderFileUpload.tsx:80]: 🙌🏻

File: services/common/src/components/forms/RenderFileUpload.tsx [hunk 5]
```diff
@@ -585,12 +592,4 @@ export const FileUpload = (props: FileUploadProps) => {
...
-const mapDispatchToProps = (dispatch) =>
-  bindActionCreators(
-    {
-      pollDocumentUploadStatus,
-    },
-    dispatch
-  );
-
-export default withFeatureFlag(connect(null, mapDispatchToProps)(FileUpload));
+export default FileUpload;
```
Comment (matbusby-fw) [inline services/common/src/components/forms/RenderFileUpload.tsx:595]: Nice solution on this file as a whole!  Happy to see some of the clean up too!

File: services/common/src/redux/createAppSlice.ts (hunk 1 above)
Comment (taraepp) [reply services/common/src/redux/createAppSlice.ts:10]: This one yeah. We should probably have a _better_ handler for errors, but it's what we're currently doing in slices and I thought it better to at least make it a _shared_ function.

File: services/document-manager/backend/Dockerfile (hunk 2 above)
Comment (matbusby-fw) [reply services/document-manager/backend/Dockerfile]: I was working with this in my PR too, but from what I can tell the folder doesn't get persisted to the resulting container after this command (/tmp/celery also doesn't exist if you exec into the container after creation)

Comment (matbusby-fw) [reply services/document-manager/backend/Dockerfile]: In my PR I added the creation in the init file for docman.

**Review by matbusby-fw — DISMISSED** [review]
Looks great.  A couple thoughts/suggestions, but it's looking nice and clean!

File: services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx (hunk 3 above)
Comment (taraepp) [reply services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx]: Agreed. Extracted a function!

File: services/document-manager/backend/Dockerfile (hunk 2 above)
Comment (taraepp) [reply services/document-manager/backend/Dockerfile]: I just took this stuff out from both Dockerfiles. 👍

**Review by matbusby-fw — APPROVED** [review]

File: services/common/src/components/documents/DocumentCompression.tsx [hunk 6]
```diff
@@ -13,18 +11,23 @@ import DocumentCompressionWarningModal from "./DocumentCompressionWarningModal";
 import DocumentCompressedDownloadModal from "./DocumentCompressedDownloadModal";
 
 interface DocumentCompressionProps {
-  documentType: string;
```
Comment (asinn134) [inline services/common/src/components/documents/DocumentCompression.tsx:16]: Thanks for cleaning up this file.

**Review by simensma-fresh — APPROVED** [review]

**Pull request merged**